### DISCUSSION TITLE
Not auto merging pin PR

### DISCUSSION THREAD

**Discussion by richardsimko** [renovatebot/renovate#8362] [Request Help]
I have the following config in one of my repos, however it's still not auto merging when tests are complete.

```
{
  "extends": [
    "config:js-app"
  ],
  "labels": ["dependencies"],
  "reviewers": ["richardsimko", ...],
  "reviewersSampleSize": 2,
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    }
  ]
}
```

The log gives me:

```
DEBUG: PR is configured for automerge(branch="renovate/pin-dependencies")
DEBUG: PR is not ready for merge(branch="renovate/pin-dependencies")
{
  "canMergeReason": "mergeStateStatus = UNSTABLE"
}
```

However I'm not sure what `UNSTABLE` means? All the statuses are reported as OK in GH:

![image](https://user-images.githubusercontent.com/1245031/105177131-8862e700-5b26-11eb-8eb6-7d2dde89bd20.png)

Is the bot supposed to react to the status change on its own?

**Comment by rarkins** [comment]
It's a GitHub term, but they don't give too much details: https://docs.github.com/en/graphql/reference/enums#mergestatestatus

Are you sure that Renovate has run *since* all the checks passed?

**Reply by richardsimko** [reply]
No, it seems to run on changes to master and rebase the branch, after which the tests run and Renovate doesn't seem to re-run. Is it re-run on a schedule or is it supposed to trigger off a webhook? I'v waited about 30 minutes and it hasn't re-run. 🤔

**Reply by rarkins** [reply]
Renovate is scheduled to run hourly per repo, plus whenever it gets a relevant webhook. It *does not* run instantly that a branch turns green, so you will need to wait up to an hour for the automerge. Alternatively, enable the dependencyDashboard feature and there's a checkbox you can tick there to manually trigger a run.

**Reply by richardsimko** [reply]
Ah, I'll try waiting an hour then :) I thought it would trigger from both commits and build statuses. I'll look into dependencyDashboard, thanks!

**Comment by richardsimko** [comment]
For the latest run now I got

```
DEBUG: PR is configured for automerge(branch="renovate/pin-dependencies")
DEBUG: getBranchStatus(renovate/pin-dependencies)(branch="renovate/pin-dependencies")
DEBUG: branch status check result(branch="renovate/pin-dependencies")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/pin-dependencies")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
DEBUG: PR is not ready for merge (branch status is yellow)
```

Which is odd since there should be statuses reported right away (GH Actions for example)

**Reply by rarkins** [reply]
Did it rebase/re-commit on this recent run?

**Reply by richardsimko** [reply]
Yep, it rebased some changes from master

**Reply by rarkins** [reply]
It will not automerge in the same run as it commits or rebases. You need your master to be stable enough that when Renovate runs the next time it finds that the automerging PR is (a) up to date with master, and (b) passed all test results

**Comment by rarkins** [comment]
@HonkingGoose do you think we need to add even more clarifying descriptions to our automerge doc? Please see above

**Reply by richardsimko** [reply]
Yeah maybe it's just me but this sentence made me presume that it would trigger from the status turning green:

> Automerging defaults to using Pull Requests (automergeType="pr"). In that case Renovate first creates a branch, then an associated Pull Request, and then automerges the first time it detects that the Pull Requests status checks are "green".

**Reply by HonkingGoose** [reply]
@rarkins Yeah the current https://github.com/renovatebot/renovate/blob/master/docs/usage/automerge-configuration.md document is not really helpful if you're not intimately familiar with the Renovate bot. It's also lacking a kind of order of operations.

Maybe it would be good to have a "story" of a user who wants to enable automerge on GitHub, and go through all the steps one by one. By having it in a story format, we can follow a logical order, and insert explanatory texts where needed. This would also help the new unfamiliar concepts stick, by seeing them applied.
By making a "story" for automerge on GitHub, we probably cover the most used platform?

**Reply by viceice** [reply]
Maybe we should add some sub pages for platform specific branch protection settings?

**Comment by richardsimko** [comment]
It merged on the next run! Thanks for the amazing support @rarkins 🎉
